doc/user/content/ops/deployment.md

```diff
@@ -12,61 +12,135 @@ issue](https://github.com/MaterializeInc/materialize/issues/new?labels=C-feature
 
 ## Memory
 
-Materialize stores the majority of its state in-memory, and works best when the streamed data
-can be reduced in some way. For example, if you know that only a subset of your rows and columns
-are relevant for your queries, it helps to avoid materializing sources or views until you've
-expressed this to the system (we can avoid stashing that data, which can in some cases dramatically
-reduce the memory footprint).
-
-To minimize the chances that Materialize runs out of memory in a production environment,
-we recommend you make additional memory available to Materialize via a SSD-backed
-swap file or swap partition.
-
-This is particularly important in Linux and in Docker, where swap may not be automatically
-setup for you:
+Materialize stores the majority of its state in memory, and works best when the
+streamed data can be reduced in some way. For example, if you know that only a
+subset of your rows and columns are relevant for your queries, it helps to avoid
+materializing sources or views until you've expressed this to the system.
+Materialize can then avoid stashing the full set of rows and columns, which can
+in some cases dramatically reduce Materialize's memory footprint.
+
+### Compaction
+
+To prevent memory from growing without bound, Materialize periodically
+"compacts" data in [arrangements](/overview/api-components#indexes). For
+example, if you have a source that tracks product inventory, you might receive
+periodic inventory updates throughout the day:
+
+```
+(T-shirts, 9:07am, +500)
+(T-shirts, 11:32am, -1)
+(T-shirts, 3:14pm, -2)
+```
+
+After logical compaction, Materialize will store only the current inventory
+state:
+
+```
+(T-shirts, 3:14pm, +497)
+```
+
+Materialize will only perform this compaction on data that falls outside the
+logical compaction window. The default compaction window is 60 seconds, but the
```

Comment:
I think that the delay is only 60 seconds iff timestamps advance at roughly once per millisecond? All the code that determines the compaction frontier eventually converts the compaction window arg to milliseconds, but this means that for say a BYO source where timestamps advance once per hour, we'll retain the past 60,000 hours of historical data.

Reply:
Totally accurate. I'm not sure the best way to communicate it though. We have elsewhere (in the system) a bunch of conflation of timestamps with integers in footgun fashion (you can join BYO sources with logging / catalog sources, and nothing will work).

Maybe "the default compaction window for sources with millisecond timestamps is 60 seconds"?

Reply:
IMO we should expose a parameter thats more like "number of old timestamps we will keep around" (name tbd) instead of expressing this as a time bound at all? and then i think we can talk about timestamp rates (kind of oxymoronical) and how this reduces to a compaction lag in time more naturally

Reply:
Since we only have millisecond sources now I'm going to leave as is. We can reword when we expose CDCv2 sources that have more custom timestamps!